go/tools/builders/env.go

```diff
@@ -133,18 +136,20 @@ func (e *env) runCommand(args []string) error {
 	cmd := exec.Command(args[0], args[1:]...)
 	// Redirecting stdout to stderr. This mirrors behavior in the go command:
 	// https://go.googlesource.com/go/+/refs/tags/go1.15.2/src/cmd/go/internal/work/exec.go#1958
-	cmd.Stdout = os.Stderr
-	cmd.Stderr = os.Stderr
-	return runAndLogCommand(cmd, e.verbose)
+	var stderr bytes.Buffer
+	cmd.Stdout = &stderr
+	cmd.Stderr = &stderr
+	return runAndLogCommand(cmd, &stderr, e.verbose)
 }
 
 // runCommandToFile executes a subprocess and writes the output to the given
 // writer.
 func (e *env) runCommandToFile(w io.Writer, args []string) error {
 	cmd := exec.Command(args[0], args[1:]...)
+	var stderr bytes.Buffer
```

Comment:
Let's make sure this function doesn't change. It needs to produce a file, and we shouldn't alter the content of that file.

Maybe some other function can call `runAndLogCommand` with a `*bytes.Buffer`, then `relativizePaths` on that and copy to `os.Stderr`? Then neither `runAndLogCommand` nor this function need to change.

Reply:
This is a lot cleaner, very good idea!